src/test/java/org/folio/innreach/controller/InnReachLocationControllerTest.java

```diff
@@ -0,0 +1,145 @@
+package org.folio.innreach.controller;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import static org.folio.innreach.fixture.TestUtil.deserializeFromJsonFile;
+
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.jdbc.Sql;
+
+import org.folio.innreach.controller.base.BaseControllerTest;
+import org.folio.innreach.dto.InnReachLocationDTO;
+import org.folio.innreach.dto.InnReachLocationsDTO;
+
+class InnReachLocationControllerTest extends BaseControllerTest {
+
+	private static final String PRE_POPULATED_LOCATION1_ID = "a1c1472f-67ec-4938-b5a8-f119e51ab79b";
+	private static final String PRE_POPULATED_LOCATION2_ID = "26f7c8c5-f090-4742-b7c7-e08ed1cc4e67";
+	private static final String PRE_POPULATED_LOCATION3_ID = "34c6a230-d264-44c5-90b3-6159ed2ebdc1";
+
+	@Autowired
+	private TestRestTemplate testRestTemplate;
+
+	@Test
+	void return200HttpCode_and_createdInnReachLocation_when_createInnReachLocation() {
+		var innReachLocationDTO = deserializeFromJsonFile("/inn-reach-location/create-inn-reach-location-request.json",
+				InnReachLocationDTO.class);
+
+		var responseEntity = testRestTemplate.postForEntity("/locations", innReachLocationDTO,
```

Comment:
Please pay attention for route path, if it will be changed in definition tests will not work.